### PR TITLE
Add spelling check script and fix typos

### PR DESCRIPTION
**Pull request by islinwb**
/assign @m1093782566

### CODE REVIEW

File: hack/.spelling_failures [hunk 1]
```diff
@@ -0,0 +1,4 @@
+BUILD
+OWNERS
+third_party/
+vendor/
```
Comment (m1093782566) [inline hack/.spelling_failures]: A newline please.

File: hack/verify-spelling.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+PATH="${GOBIN}:${PATH}"
+
+# Install tools we need, but only from vendor/...
+go install k8s.io/kubernetes/vendor/github.com/client9/misspell/cmd/misspell
+
+# Spell checking
+# All the skipping files are defined in hack/.spelling_failures
+skipping_file="${KUBE_ROOT}/hack/.spelling_failures"
+failing_packages=$(echo `cat ${skipping_file}` | sed "s| | -e |g")
+git ls-files | grep -v -e ${failing_packages} | xargs misspell -i "Creater,creater,ect" -error -o stderr
```
Comment (m1093782566) [inline hack/verify-spelling.sh]: A newline please.

**Comment by m1093782566** [comment]
```
W1012 07:26:55.525] can't load package: package k8s.io/kubernetes/vendor/github.com/client9/misspell/cmd/misspell: cannot find package "k8s.io/kubernetes/vendor/github.com/client9/misspell/cmd/misspell" in any of:
W1012 07:26:55.526] 	/usr/local/go/src/k8s.io/kubernetes/vendor/github.com/client9/misspell/cmd/misspell (from $GOROOT)
W1012 07:26:55.526] 	/go/src/k8s.io/kubernetes/vendor/github.com/client9/misspell/cmd/misspell (from $GOPATH)
```

**Comment by islinwb** [comment]
/hold
Let me fix the failed test first...

**Comment by m1093782566** [comment]
Error logs of test:

```
I1015 03:21:07.530] +++ [1015 03:21:07] Creating data container kube-build-data-6bae5ecc2a-5-v1.9.3-1
I1015 03:21:18.250] +++ [1015 03:21:18] Syncing sources to container
I1015 03:21:26.381] +++ [1015 03:21:26] Running build command...
I1015 03:21:27.206] Skipping generated files....
I1015 03:21:27.710] +++ [1015 03:21:27] Building the toolchain targets:
I1015 03:21:27.711]     github.com/kubernetes-sigs/poseidon/vendor/k8s.io/kubernetes/hack/cmd/teststale
I1015 03:21:27.767] Makefile:92: recipe for target 'all' failed
I1015 03:21:27.770] Makefile:442: recipe for target 'cross' failed
W1015 03:21:27.870] can't load package: package github.com/kubernetes-sigs/poseidon/vendor/k8s.io/kubernetes/hack/cmd/teststale: cannot find package "github.com/kubernetes-sigs/poseidon/vendor/k8s.io/kubernetes/hack/cmd/teststale" in any of:
W1015 03:21:27.871] 	/usr/local/go/src/github.com/kubernetes-sigs/poseidon/vendor/k8s.io/kubernetes/hack/cmd/teststale (from $GOROOT)
W1015 03:21:27.871] 	/go/src/github.com/kubernetes-sigs/poseidon/_output/dockerized/go/src/github.com/kubernetes-sigs/poseidon/vendor/k8s.io/kubernetes/hack/cmd/teststale (from $GOPATH)
W1015 03:21:27.871] !!! [1015 03:21:27] Call tree:
W1015 03:21:27.871] !!! [1015 03:21:27]  1: /go/src/github.com/kubernetes-sigs/poseidon/hack/lib/golang.sh:616 kube::golang::build_kube_toolchain(...)
W1015 03:21:27.871] !!! [1015 03:21:27]  2: hack/make-rules/build.sh:27 kube::golang::build_binaries(...)
W1015 03:21:27.871] !!! [1015 03:21:27] Call tree:
W1015 03:21:27.871] !!! [1015 03:21:27]  1: hack/make-rules/build.sh:27 kube::golang::build_binaries(...)
W1015 03:21:27.871] make[1]: *** [all] Error 1
W1015 03:21:27.872] make: *** [cross] Error 1
W1015 03:21:27.991] !!! [1015 03:21:27] Call tree:
W1015 03:21:27.993] !!! [1015 03:21:27]  1: build/../build/common.sh:551 kube::build::run_build_command_ex(...)
W1015 03:21:27.995] !!! [1015 03:21:27]  2: build/release.sh:36 kube::build::run_build_command(...)
W1015 03:21:27.996] make: *** [quick-release] Error 1
```

**Comment by m1093782566** [comment]
/retest

hope #143 can help the tests pass.

**Comment by shivramsrivastava** [comment]
/lgtm

Thanks @islinwb and @m1093782566

**Pull request merged**